### DISCUSSION TITLE
Get the corrected pose on ros topic

### DISCUSSION THREAD

**Discussion by Paul-Ajay** [OpenVSLAM-Community/openvslam_ros#160] [Q&A] [answered]
Currently the camera pose published on the `camera_pose` does not account the keyframes after pose graph optimization. Because of this the sudden jumps are observing on this topic. Is there any way to get the corrected pose on the topic?

**Comment by ymd-stella** [comment]
> Currently the camera pose published on the camera_pose does not account the keyframes after pose graph optimization. Because of this the sudden jumps are observing on this topic.

Provide additional information.

**Reply by Paul-Ajay** [reply]
The trajectory of camera shown in the PangolinViewer is this after correcting the pose graph because of loop closure. 
![stella_mono_pangolin](https://github.com/stella-cv/stella_vslam_ros/assets/20149093/5db5a446-6881-4d00-b9f7-705a8fae1f0d)
But when I plot the topic `camera_pose`, the loop closure and pose graph optimization is not accounted and we can see a jump in camera pose.
![stella_mono](https://github.com/stella-cv/stella_vslam_ros/assets/20149093/691e2e15-fb71-456c-9b2f-885bc1b24283)

**Reply by ymd-stella** [reply, marked as answer]
How about specifying `--eval-log-dir` and using the log?

**Reply by Paul-Ajay** [reply]
Thanks, that helped to get the keyframe poses in required dataformat.
But the timestamp in frame & keyframe trajectory files are starting from zero. Is there any way to change to system time ?
[frame_trajectory.txt](https://github.com/stella-cv/stella_vslam_ros/files/12533774/frame_trajectory.txt)

**Reply by ymd-stella** [reply]
It matches the timestamp of the image message.

**Reply by Paul-Ajay** [reply]
Thanks. I was using the ros branch which doesnt have this change accommodated https://github.com/stella-cv/stella_vslam_ros/blob/ros/src/stella_vslam_ros.cc#L238. Refering the ros2 branch https://github.com/stella-cv/stella_vslam_ros/blob/ros2/src/stella_vslam_ros.cc#L226C5-L226C17 and correcting it solved the issue
